### PR TITLE
use domain when solving inequalities

### LINKED ISSUE #10477: sympy.inequalities.solve_univariate_inequality incorrectly solves rational inequalities
The function `sympy.inequalities.solve_univariate_inequality` does not return the correct solution for rational inequalities. For example:

``` python
>>> import sympy
>>> from sympy.abc import x
>>> sympy.inequalities.solve_univariate_inequality((x**2+4*x-3)/x<2,x)
And(-oo < x, x < -3)
```

which is not the complete  solution ( complete solution is x<-3 and 0<x<=1). 
I have tried different other rational inequalities and it always return incomplete solution.

Curiously the complementary inequality is solved almost correctly. It appears it does not take into account the sign change at the pole.

``` python
>>> sympy.inequalities.solve_univariate_inequality((x**2+4*x-3)/x>2,x)
Or(And(-3 < x, x < 1), And(1 < x, x < oo))
```

The versions:

```
>>> sympy.__version__
'0.7.6.1'
>>> sys.version
'3.4.3+ (default, Oct 14 2015, 16:03:50) \n[GCC 5.2.1 20151010]'
```

### LINKED ISSUE #10671: solveset(sin(y), y, Interval(0, pi)) is wrong
```
In [5]: solveset(sin(y), y, Interval(0, pi))
Out[5]: {2⋅n⋅π | n ∊ ℤ} ∪ {2⋅n⋅π + π | n ∊ ℤ}
```

It isn't intersecting the domain `Interval(0, pi)`. 

@smichr

### LINKED ISSUE #11064: solveset(x+sqrt(x**2-5)>0,x,S.Reals) gives wrong resullt
In [4]: solveset(x+sqrt(x**2-5)>0,x,S.Reals)
Out[4]: ∅

Result should be [sqrt(5), ∞)

### LINKED ISSUE #12429: solveset(log(x)/x <= 0, x, S.Reals) ->(0, oo) instead of (0, 1]
```
>>> var('x')
x
>>> solveset(log(x)/x<=0,x,S.Reals)
(0, oo)
```

The solution should be `(0, 1]`.

There is, perhaps, some issue with the `is_finite_with_finite_vars` routine which behaves as follows:

```
>>> from sympy.solvers.solveset import _is_finite_with_finite_vars as f
>>> f(1/Dummy())
True
>>> f(1/Dummy(real=True))
>>> f(1/Dummy(complex=True))
True
```

This is related to the reason behind why the test is failing that PR #12422 is addressing.

### LINKED ISSUE #12466: Wrong result in solveset function
On using solveset function to calculate this
`>>> solveset((log(x-6)/x)<=0,x,S.Reals)`
`{7}`
I think output should be (6,7], as the lhs is less than or equal to 0 for (6,7]. 

For examples such as
`>>> solveset(log(x+5)/x<=0,x,S.Reals)`
`[−4,0)`
Output is correct.

**My Observations** (a,b are constants)
Functions as log(x + a)/b<=0. Output is correct.
Functions as      log(x - a)/b<=0. Output is wrong

I have also mentioned this issue in comments on post [https://github.com/sympy/sympy/issues/12429](https://github.com/sympy/sympy/issues/12429). 

Thank you.

### LINKED ISSUE #12482: is_finite_with_finite_vars output depends on symbol assumptions
`is_finite_with_finite_vars` routine behaves as follows:

```
>>> from sympy.solvers.solveset import _is_finite_with_finite_vars as f
>>> f(1/Dummy())
True
>>> f(1/Dummy(real=True))
>>> f(1/Dummy(complex=True))
True
```

This is related to the reason behind why the test is failing that PR #12422 is addressing.
This was copied from #12429 as a related issue.

### PR DESCRIPTION
**Pull request by smichr**
- a test for the original issue was added
- a test for the new issue raised within the same issue
  was added and other tests that were wrong were updated
- the calculation of invalid solutions was removed since
  this is already handled in the solution of the inequality

closes #10671 
closes #12466
closes #10477
closes #11064
closes #12429
closes #12482
closes #12422
closes #12458

### CODE REVIEW

File: sympy/calculus/singularities.py [hunk 1]
```diff
@@ -50,13 +51,18 @@ def singularities(expression, symbol):
...
         raise NotImplementedError(
             "Algorithms finding singularities for non-rational"
             " functions are not yet implemented."
         )
     else:
-        return solveset(simplify(1 / expression), symbol)
+        singularities = []
+        for d in denoms(expression):
+            singularities.extend(solveset(d, symbol))
+        return FiniteSet(*singularities)
```
Comment (kshitij10496) [inline sympy/calculus/singularities.py]: @smichr I think that we should make a note of the fact that this function returns only the isolated singularities of the given expression. ie. we need to state that non-isolated singularities and branch points are not determined by this function.

Comment (smichr) [inline sympy/calculus/singularities.py]: done

**Comment by maurogaravello** [comment]
This corrects also the #12466.

**Comment by maurogaravello** [comment]
Seems good for me.

**Comment by smichr** [comment]
@maurogaravello , would you be willing to take a look at the commits that come after your last review?

**Comment by maurogaravello** [comment]
I am thinking about the changes. A first impression is that making the integral test more robust is necessary. It seems to me that `assert Sum(-ln(n)/n**3, (n, 1, oo)).is_convergent() is S.true` gives a failure. The sum is clearly convergent, but `-ln(n)/n**3` is not decreasing when `n` is big. 
Mathematically speaking, the correct assumption should be the decreasing of abs(-ln(n)/n**3). 
I tried to implement this check some days ago, but sympy seems not to be able to do in all situations.

So a possibility for `Sum(expr, (n, 1, oo)).is_convergent()` could be: checking if expr is decreasing (if it is positive) or checking if -expr is decreasing (if it is negative)

**Comment by smichr** [comment]
> but -ln(n)/n**3 is not decreasing when n is big.

True: it is increasing to 0. That is the purpose of the two tests (which essentially test `abs(f(x))`. If `f(x)` is increasing towards zero past its last maxima then `-f(x)` is *decreasing* towards 0 past the same maxima.

**Comment by maurogaravello** [comment]
Sorry. I did not see line 494. 
I agree with your proposal.

**Comment by smichr** [comment]
Do you want to cherry pick to your branch? I would send you a pr, but I can't right now.

**Comment by smichr** [comment]
> remove NotImplementedError from singularities

I wonder if singularities should raise an error if anything but a finite set of expressions is obtained.

**Comment by maurogaravello** [comment]
No problem with the PR. Your solution is better than mine.

Should I delete my PR?

**Comment by smichr** [comment]
> Should I delete my PR?

I prefer to attribute your work to you. If you hadn't done what you did I don't know if I would have done what I did. If it's too much of a hassle we can just keep the changes here.

**Comment by smichr** [comment]
Test is split 4 needs some debugging:

```
6149____________ sympy/integrals/tests/test_integrals.py:test_diff_wrt _____________

6150  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/integrals/tests/test_integrals.py", line 107, in test_diff_wrt

6151    assert integrate(t + 1, (t, 0, 1)) == S(3)/2

6152  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/integrals/integrals.py", line 1295, in integrate

6153    risch=risch, manual=manual)

6154  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/integrals/integrals.py", line 553, in doit

6155    evalued = Add(*others)._eval_interval(x, a, b)

6156  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/core/expr.py", line 818, in _eval_interval

6157    _singularities = singularities(self, x, include_removable=False)

6158  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/calculus/singularities.py", line 69, in singularities

6159    for d in denoms(expression, symbol):

6160  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy/solvers/solvers.py", line 127, in denoms

6161    if any(s in free for s in symbols):

6162TypeError: 'Test' object is not iterable

```

`Test` is created for the purpose of the test in the suite. I'm not sure why it is being asked to iterate.

**Comment by smichr** [comment]
And `sets` tests appear to be hanging. Also `sympy/combinatorics/tests/test_partitions.py`?

**Comment by kshitij10496** [comment]
> I wonder if singularities should raise an error if anything but a finite set of expressions is obtained.

I think this will restrict the use of this functionality(`singularity`) to non-trigonometric/non-periodic functions. For example, expressions with countably infinite number of disconitnuites such as`1/cos(x)` wouldn't be correctly computed by this method in the future.
What do you think about this ?

**Comment by smichr** [comment]
> What do you think about this ?

I'm open to suggestions. I guess the easiest thing to do is just return the sympy-state-of-the-art solveset solution for the singularities and the user will make up their own mind about what to do with them. Any idea why that rational restriction was there in the first place?

**Comment by smichr** [comment]
When tests pass, I will commit unless I hear something more from @parsoyaarihant . I removed changes to `singularities` except for the note suggested (I believe) by @kshitij10496

**Pull request merged**